src/engine/engine_common.c

```diff
@@ -432,13 +425,152 @@ static int lock_clines(struct ocf_request *req,
 	}
 }
 
+static inline int ocf_evict_part_overflow(struct ocf_request *req,
+		const struct ocf_engine_callbacks *engine_cbs, uint32_t clines_to_evict)
+{
+	int lock_status = -OCF_ERR_NO_LOCK;
+	struct ocf_metadata_lock *metadata_lock = &req->cache->metadata.lock;
+
+	ocf_metadata_start_exclusive_access(metadata_lock);
+
+	if (space_managment_evict_do(req->cache, req, clines_to_evict) ==
+			LOOKUP_MISS) {
+		goto unlock;
+	}
+
+	if (!ocf_part_is_enabled(req)) {
+		req->info.mapping_error = 1;
+		goto unlock;
+	}
+
+	lock_status = lock_clines(req, engine_cbs);
+
+unlock:
+	ocf_metadata_end_exclusive_access(metadata_lock);
+
+	return lock_status;
+}
+
+static inline int ocf_prepare_clines_hit(struct ocf_request *req,
+		const struct ocf_engine_callbacks *engine_cbs)
+{
+	int lock_status = -OCF_ERR_NO_LOCK;
+	uint32_t clines_to_evict;
+	int res;
+
+	/* Cachelines are mapped in correct partition */
+	if (!ocf_engine_needs_repart(req)) {
```

Comment:
what if target partition is disabled? shouldn't we attempt to evict from it here?